gdk/commands/component/build.py

```diff
@@ -317,13 +317,26 @@ def copy_artifacts_and_update_uris():
                         f"Could not find the artifact file specified in the recipe '{artifact_file}' inside the build folder"
                         f" '{build_folder}'."
                     )
-            if not artifact_found:
+            logging.warning(f"Could not find the artifact file '{artifact_file}' in the build folder '{build_folder}'.")
+            # Check for the artifact in s3 as well before raising an exeception
+            if not artifact_found and not is_artifact_in_s3(artifact["URI"]):
                 raise Exception(
-                    f"Could not find the artifact file specified in the recipe '{artifact_file}' inside the build folders."
-                    f" '{build_folders}'."
+                    f"Could not find artifact with URI '{artifact_uri}' on s3 or inside the build folders '{build_folders}'."
                 )
 
 
+def is_artifact_in_s3(artifact_uri):
+    bucket_name, object_key = artifact_uri.replace(utils.s3_prefix, "").split("/", 1)
+    try:
+        s3_client = project_utils.create_s3_client(project_config["region"])
```

Comment:
How costly is client creation? Is it cached?

Reply:
Good catch. Right now, client is created for every artifact that's not found in the local build folder. Need to revise the code.